src/test/http_rest_api_handler_test.cpp

```diff
@@ -205,3 +214,252 @@ TEST(ModelControlApi, reloadNotNeededManyThreads) {
+
+    auto handler = ovms::HttpRestApiHandler(10);
+    std::string response;
+
+    ovms::ModelManager& manager = ovms::ModelManager::getInstance();
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+    manager.loadConfig(configFile);
+    std::filesystem::remove("/tmp/ovms_config_file.json");
+    createConfigFileWithContent(config_with_pipeline);
+    std::this_thread::sleep_for(std::chrono::seconds(1));
+
+    const char* expectedJson = R"("models" : 
+{
+"dummy" : 
+{
+ "model_version_status": [
+  {
+   "version": "1",
+   "state": "AVAILABLE",
+   "status": {
+    "error_code": "OK",
+    "error_message": "OK"
+   }
+  }
+ ]
+}
+},
+"pipelines" : 
+{
+ "pipeline1Dummy" : "AVAILABLE"
+})";
+    auto status = handler.processConfigReloadRequest(response);
+
+    EXPECT_EQ(expectedJson, response);
+    EXPECT_EQ(status, ovms::StatusCode::OK_CONFIG_FILE_RELOAD_NEEDED);
+}
+
+static const char* config_with_pipelines = R"(
+{
+    "model_config_list": [
+        {
+            "config": {
+                "name": "dummy",
+                "base_path": "/ovms/src/test/dummy"
+            }
+        }
+    ],
+    "pipeline_config_list": [
+        {
+            "name": "pipeline1Dummy",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                            "data_item": "custom_dummy_input"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "a",
+                        "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                        "data_item": "new_dummy_output"}
+                }
+            ]
+        },
+        {
+            "name": "pipeline2Dummy",
+            "inputs": ["custom_dummy_input"],
+            "nodes": [
+                {
+                    "name": "dummyNode",
+                    "model_name": "dummy",
+                    "type": "DL model",
+                    "inputs": [
+                        {"b": {"node_name": "request",
+                            "data_item": "custom_dummy_input"}}
+                    ], 
+                    "outputs": [
+                        {"data_item": "a",
+                        "alias": "new_dummy_output"}
+                    ] 
+                }
+            ],
+            "outputs": [
+                {"custom_dummy_output": {"node_name": "dummyNode",
+                                        "data_item": "new_dummy_output"}
+                }
+            ]
+        }
+    ]
+})";
+
+TEST(ModelControlApi, configChangeWithPipelines) {
+    std::filesystem::remove("/tmp/ovms_config_file.json");
```

Comment:
You are inheriting from TestWithTempDir fixture so why you are not reusing SetUp()/TearDown() methods?